### PR TITLE
Ports full auto guns

### PR DESCRIPTION
**Pull request by TheNeoGamer42**
## About The Pull Request
ports https://github.com/BeeStation/BeeStation-Hornet/pull/1702 into boomer, after several hours of pain

## Why It's Good For The Game
full auto guns, what the fuck else do you expect, you can fucking shoot into a crowd of people with an L6 and not murder your finger

## Changelog
:cl:
add: The L666 FUCKYOUINATOR, a modified L6 Saw that is admin only because its overpowered.
tweak: holy shit guns shoot good
tweak: I think it changed some damage values and shit too
add: new fire rate var for guns
tweak: phases out burstfire on guns that don't specifically need it in favor of true automatic fire
tweak: universal fire rate changes
code: fire rate is now tied to gun, and not ammo casing
balance: lowered energy cost of temp gun projectiles by 60%
/:cl:

### CODE REVIEW

**Comment by TheNeoGamer42** [comment]
actually gonna copy bits of the changelog from the bee pr that I know I kept

**Comment by TheNeoGamer42** [comment]
fuck you github bot, it worked for my fucking code

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 1]
```diff
@@ -95,7 +97,9 @@
 		if(prob(100 - (magazine.ammo_count() * 5)))	//minimum probability of 70, maximum of 95
 			playsound(user, fire_sound, fire_sound_volume, vary_fire_sound)
 			to_chat(user, "<span class='userdanger'>[src] blows up in your face!</span>")
-			user.gib()
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/guns/ballistic/revolver.dm:98]: !!!

**Comment by triplezeta** [comment]
haha autorifles go BRRR

File: code/modules/projectiles/guns/ballistic/revolver.dm (hunk 1 above)
Comment (TheNeoGamer42) [reply code/modules/projectiles/guns/ballistic/revolver.dm:98]: fiiiiiiiinnnnnnnneeeeeeeee

**Comment by TheNeoGamer42** [comment]
hjonk

**Comment by TheNeoGamer42** [comment]
fucking work now, goddamnet

**Comment by TheNeoGamer42** [comment]
marg,,,, test merg,,,,,,

**Comment by MarkSuckerberg** [comment]
Doesn't seem to runtime, and nobody's complained ~~(yet)~~,  so I'll merge it soon.

**Comment by dinnerwoerror** [comment]
so does this just buff syndicate weapons even further, considering how pretty much every gun in the game except for pistols, armory guns, and the moist nugget are now fully automatic

**Comment by TheNeoGamer42** [comment]
I am probably gonna make some firerate changes, mainly to the autorifle since peopel anger about its garbage firerate

**Comment by TheNeoGamer42** [comment]
I might also strip the burst from the syndicate guns and increase the firerate, they fire so fastright now because they still burst fire

**Comment by dinnerwoerror** [comment]
however i am defenitely liking the temperature gun upgrade

got a lot of shots indeed

**Comment by TheNeoGamer42** [comment]
mmm yes nerf tiem

**Comment by dinnerwoerror** [comment]
and by that do you mean nerfing all of the fully automatic guns or do you mean nerfing the fully automatic guns that aren't available to non-antagonists

**Comment by TheNeoGamer42** [comment]
nerf the code by rebasing to nocode

**Comment by TheNeoGamer42** [comment]
the only automatic gun the crew has access to is the autorifle and its uh... kinda garbage last time I checked

**Comment by TheNeoGamer42** [comment]
I still want actual proper fucking feedback on what the various firerates should be

**Comment by dinnerwoerror** [comment]
i personally still think that we should just keep the old burst fire rates since this just seems like another antag buff

**Comment by TheNeoGamer42** [comment]
well I can just strip the burst which is currently making the syndicate guns actually seem pretty good firerates

**Comment by TheNeoGamer42** [comment]
also it technically nerfs antags in some aspects too, the syndicate revolver fires slower and a few other changes

**Comment by dinnerwoerror** [comment]
is this what we get when we put the L6 into the pack a punch machine

**Comment by TheNeoGamer42** [comment]
are you talking about the L666? because the L666 is just.... it

**Comment by MarkSuckerberg** [comment]
It's not a question of if I'll regret this.

It's a question of when.

**Comment by TheNeoGamer42** [comment]
oh god you merged it

**Comment by TheNeoGamer42** [comment]
time to make balance PRs afterwards

**Pull request merged**